Shared/Samples/Find route in mobile map package/FindRouteInMobileMapPackageView.MobileMapView.swift

```diff
@@ -0,0 +1,166 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+extension FindRouteInMobileMapPackageView {
+    /// The map view for a mobile map package map.
+    struct MobileMapView: View {
+        /// The view model for the view.
+        @StateObject private var model: Model
+        
+        /// The placement of the address callout on the map.
+        @State private var calloutPlacement: CalloutPlacement?
+        
+        /// The text address shown in the callout.
+        @State private var calloutText: String = ""
+        
+        /// The point on the screen where the user tapped.
+        @State private var tapScreenPoint: CGPoint?
+        
+        /// A Boolean value indicating whether the reset button is disabled.
+        @State private var resetDisabled = true
+        
+        /// A Boolean value indicating whether the error alert is showing.
+        @State private var errorAlertIsShowing = false
+        
+        /// The error shown in the error alert.
+        @State private var error: Error? {
+            didSet { errorAlertIsShowing = error != nil }
+        }
+        
+        init(map: Map, locatorTask: LocatorTask) {
+            let model = Model(map: map, locatorTask: locatorTask)
+            _model = StateObject(wrappedValue: model)
+        }
```

Comment:
Instead of passing the `map` and `locatorTask` to create the view model for the `MobileMapView`, I think it would be better to pass the mobile map package to the child view.

- The list view only needs to display the metadata of an mmpk - its item name and thumbnail (which is a `LoadableImage`), so the mmpks can take up less memory
- When a `MobileMapView` is shown, we then load the package and display the map in it, instantiate the locator task, etc. So the memory is only consumed when a map is shown, not when the list is shown.
- Some of the mmpk loading logics can be moved into the child view's model.

Reply:
It appears that the `MobileMapPackage` needs to be loaded even to access its item name, so I kept the same structure. I did simplify the loading, though, since it was unnecessarily complex for this use case.

Reply:
Thanks. I realized that the [item](https://developers.arcgis.com/swift/api-reference/documentation/arcgis/mobilemappackage/item) of a mobile map package has to be loaded 🤦